pkg/kubelet/kubeletconfig/configfiles/util_windows.go

```diff
@@ -0,0 +1,30 @@
+//go:build windows
+// +build windows
+
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package configfiles
+
+import (
+	"path/filepath"
+)
+
+func isAbs(path string) bool {
+	// on Windows, filepath.IsAbs will not return True for paths prefixed with a slash, even
+	// though they can be used as absolute paths (https://docs.microsoft.com/en-us/dotnet/standard/io/file-path-formats).
+	return filepath.IsAbs(path) || (len(path) > 0 && (path[0] == '\\' || path[0] == '/'))
```

Comment:
is it an error in `filepath` package logic? Is there a github issue where this might have been discussed? Looks like a common, non k8s specific problem.

Reply:
From the ``filepath`` package logic itself, it seems that it won't return ``True`` for any slash-prefixed paths (e.g.: ``\foo``), since ``volumeNameLen`` would return 0 (doesn't start with a drive letter, and it's not an UNC path), which means that ``filepath.IsAbs`` returns ``False``. Though technically, according to https://docs.microsoft.com/en-us/dotnet/standard/io/file-path-formats , it should be considered as absolute. According to the same documentation, ``\foo`` would be normalized to ``C:\foo`` (``If the path starts with a single component separator, the drive from the current directory is applied. For example, if the file path is \utilities and the current directory is C:\temp\, normalization produces C:\utilities.``). This information can also be seen here: https://learn.microsoft.com/en-us/windows/win32/fileio/naming-a-file#fully-qualified-vs-relative-paths (``Fully Qualified vs. Relative Paths - A file name is relative to the current directory if it does not begin with one of the following: ... A single backslash, for example, "\directory" or "\file.txt". This is also referred to as an absolute path.``)

It might also be worth noting that ``filepath.Abs("/foo")`` (which is a ``GetFullPathName`` syscall) will result in normalized ``C:\foo`` path, which applies the rule above.

Reply:
my question is - if this is not k8s specific issue, can we file an issue for the `filepath` package itself? Or is there any k8s-specific here?